pkgs/tools/inputmethods/touchegg/default.nix

```
@@ -1,26 +1,40 @@
-{ lib, stdenv, fetchurl, xorg, xorgserver, qt4, libGLU, libGL, geis, qmake4Hook }:
+{ stdenv, lib, fetchFromGitHub, systemd, libinput, pugixml, cairo, xorg, gtk3-x11, pcre, pkg-config, cmake }:
 
 stdenv.mkDerivation rec {
-  pname = "touchegg";
-  version = "1.1.1";
-  src = fetchurl {
-    url = "https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/touchegg/${pname}-${version}.tar.gz";
-    sha256 = "95734815c7219d9a71282f3144b3526f2542b4fa270a8e69d644722d024b4038";
+  name = "touchegg";
+  pname = "${name}";
+  version = "2.0.8";
+  src = fetchFromGitHub {
+    owner = "JoseExposito";
+    repo = "${name}";
+    rev = "f6c64bbd6d00564a980bf4379dcc1178de294c85";
+    sha256 = "1v1sskyxmvbc6j4l31zbabiklfmy2pm77bn0z0baj1dl3wy7xcj2";
   };
 
-  buildInputs = [ xorgserver libGLU libGL xorg.libX11 xorg.libXtst xorg.libXext qt4 geis ];
+  PKG_CONFIG_SYSTEMD_SYSTEMDSYSTEMUNITDIR = "${placeholder "out"}/lib/systemd/system";
 
-  nativeBuildInputs = [ qmake4Hook ];
+  buildInputs = [
+    systemd
+    libinput
+    pugixml
+    cairo
+    xorg.libX11
+    xorg.libXtst
+    xorg.libXrandr
+    xorg.libXi
+    xorg.libXdmcp
+    xorg.libpthreadstubs
+    xorg.libxcb
+    gtk3-x11
+    pcre
+  ];
 
-  preConfigure = ''
-    sed -e "s@/usr/@$out/@g" -i $(find . -name touchegg.pro)
-    sed -e "s@/usr/@$out/@g" -i $(find ./src/touchegg/config/ -name Config.cpp)
-  '';
+  nativeBuildInputs = [ pkg-config cmake ];
 
-  meta = {
+  meta = with lib; {
     homepage = "https://github.com/JoseExposito/touchegg";
-    description = "Macro binding for touch surfaces";
-    license = lib.licenses.gpl2;
-    platforms = lib.platforms.linux;
+    description = "Linux multi-touch gesture recognizer";
+    license = licenses.gpl3;
+    platforms = stdenv.lib.platforms.linux;
```

```suggestion
    platforms = platforms.linux;
```

stdenv.lib makes ofborg fail.